File: 040_Inheritance_Polymorphism/src/hr/unipu/java/inheritance_polymorphism/Beverage.java
package hr.unipu.java.inheritance_polymorphism;

public class Beverage extends Item {
	private double volume;		// Novi atribut kojeg bazna klasa nema

	// Konstruktor 0
	// -> ovdje prvoditelj automatski ubacuje super() ako se instancira s praznim konstruktorom "new Beverage()"

	// Konstruktor 1
	public Beverage(String sku, String name, double volume){
		super(sku, name);			// Prva naredba: eksplicitno poziv konstruktora bazne klase
		this.volume = volume;
	}

	// Konstruktor 2
	public Beverage(String sku, String name, double price, double volume){
		super(sku, name, price);	// Prva naredba: eksplicitno poziv konstruktora bazne klase
		this.volume = volume;
	}
	
	@Override
	public String getItemType() {
		return "Beverage";
	}

	public double getVolume() {
		return volume;
	}
	public void setVolume(double volume) {
		this.volume = volume;
	}
	
}

File: 040_Inheritance_Polymorphism/src/hr/unipu/kotlin/inheritance_polymorphism/Beverage.kt
package hr.unipu.kotlin.inheritance_polymorphism

// Izvedena klasa
class Beverage : Item {
    var volume: Double = 0.0       // Novi atribut kojeg bazna klasa nema

    override val itemType: String
        get() = "Beverage"

    // Konstruktor 0
    // -> ovdje prvoditelj automatski ubacuje super() ako se instancira s praznim konstruktorom "new Beverage()"

    // Konstruktor 1
    constructor(sku: String, name: String, volume: Double) : super(sku, name) {
        this.volume = volume
    }

    // Konstruktor 2
    constructor(sku: String, name: String, price: Double, volume: Double) : super(sku, name, price) {
        this.volume = volume
    }
}
